scripts/provision_mac.sh

```diff
@@ -23,8 +23,8 @@ export HOMEBREW_NO_AUTO_UPDATE=1
 brew install -q python cmake ninja gpg llvm@17 bash
 
 # Python 3 packages
-python3 -m pip install --user --upgrade pip
-python3 -m pip install --user -r "$SCRIPT_DIR/requirements.txt"
+python3 -m pip install --break-system-packages --user --upgrade pip
+python3 -m pip install --break-system-packages --user -r "$SCRIPT_DIR/requirements.txt"
```

Comment:
Are these still needed if this PR is just moving Linux CI now?

Reply:
No, though they'll be needed by anyone trying to port to newer macOS with newer pip. But I should remove them from this PR, good catch.

Reply:
If this PR doesn't move macOS CI over, it shouldn't touch mac-only files. That can be considered when re-enabling macOS.

Reply:
Yeah, they're definitely okay in general for another PR.  Although I'd personally prefer if we move all of our python dependency management to `uv` to make things simpler and more reliable.